### PR TITLE
Update to GNOME runtime 40

### PR DESCRIPTION
**Pull request by A6GibKm**

### CODE REVIEW

**Comment by A6GibKm** [comment]
Some module were not updated, e.g. libgweather as the compiler complained.

File: org.gnome.Calendar.json [hunk 1]
```diff
@@ -166,8 +166,8 @@
         "sources": [
             {
             "type" : "archive",
-            "url" : "https://gitlab.gnome.org/GNOME/libhandy/-/archive/v0.0.13/libhandy-v0.0.13.tar.gz",
-            "sha256" : "645355a009f23f254eaec7752b9489c3c2f5832397fcec75433a7e00efbfe52f"
+            "url" : "https://download.gnome.org/sources/libhandy/1.2/libhandy-1.2.0.tar.xz",
```
Comment (bilelmoussaoui) [inline org.gnome.Calendar.json:169]: You should drop it, it's in the sdk

Comment (A6GibKm) [reply org.gnome.Calendar.json:169]: True, tanks.

**Comment by bilelmoussaoui** [comment]
bot, build

**Comment by X-m7** [comment]
> Some module were not updated, e.g. libgweather as the compiler complained.

What was the issue here? I tried your branch plus libgweather 40.0 locally and it compiles. Granted, the weather part doesn't seem to actually work (GWeather-CRITICAL **: 09:34:32.217: gweather_info_update: assertion 'info->contact_info != NULL' failed in the terminal, and no weather info appears), but otherwise it looks fine to me.

**Comment by A6GibKm** [comment]
Ugh, it just said that meson was not supported or something like that (?), I did not look much into it to be honest. The other modules were mostly me not knowing what else required an update.

**Comment by X-m7** [comment]
You might want to try updating your flatpaks I suppose, in case there was an update to the SDK or something. Anyway, some of the other dependencies have updates too, so below is what I've changed to take care of those plus libgweather (note that libcanberra is in shared-modules now, so you'll want to update that submodule too). I'd have just suggested changes through GitHub itself, but apparently it doesn't support adding comments to unchanged lines. You should be able to use `git apply` with the patch below if you like.

```diff
diff --git a/org.gnome.Calendar.json b/org.gnome.Calendar.json
index 1b95a77..6e0278a 100644
--- a/org.gnome.Calendar.json
+++ b/org.gnome.Calendar.json
@@ -46,8 +46,8 @@
             "sources" : [
                 {
                     "type" : "archive",
-                    "url" : "https://download.gnome.org/sources/gnome-online-accounts/3.36/gnome-online-accounts-3.36.0.tar.xz",
-                    "sha256" : "1c8f62990833ca41188dbb80c5e99d99b57a62608ca675bbcd37bc2244742f2e"
+                    "url" : "https://download.gnome.org/sources/gnome-online-accounts/3.38/gnome-online-accounts-3.38.1.tar.xz",
+                    "sha256" : "e18889d67806da84d1261bfed1cf61d2baec131d2d0d0a92f83ff33d4649358e"
                 }
             ]
         },
@@ -71,8 +71,8 @@
             "sources" : [
                 {
                     "type": "archive",
-                    "url": "https://download.gnome.org/sources/libgweather/3.36/libgweather-3.36.1.tar.xz",
-                    "sha256": "de2709f0ee233b20116d5fa9861d406071798c4aa37830ca25f5ef2c0083e450"
+                    "url": "https://download.gnome.org/sources/libgweather/40/libgweather-40.0.tar.xz",
+                    "sha256": "ca4e8f2a4baaa9fc6d75d8856adb57056ef1cd6e55c775ba878ae141b6276ee6"
                 }
             ]
         },
@@ -91,22 +91,12 @@
             "sources" : [
                 {
                     "type" : "archive",
-                    "url" : "https://github.com/libical/libical/releases/download/v3.0.8/libical-3.0.8.tar.gz",
-                    "sha256" : "09fecacaf75ba5a242159e3a9758a5446b5ce4d0ab684f98a7040864e1d1286f"
-                }
-            ]
-        },
-        {
-            "name" : "libcanberra-gtk3",
-            "buildsystem" : "autotools",
-            "sources" : [
-                {
-                    "type" : "archive",
-                    "url" : "http://0pointer.de/lennart/projects/libcanberra/libcanberra-0.30.tar.xz",
-                    "sha256" : "c2b671e67e0c288a69fc33dc1b6f1b534d07882c2aceed37004bf48c601afa72"
+                    "url" : "https://github.com/libical/libical/releases/download/v3.0.9/libical-3.0.9.tar.gz",
+                    "sha256" : "bd26d98b7fcb2eb0cd5461747bbb02024ebe38e293ca53a7dfdcb2505265a728"
                 }
             ]
         },
+        "shared-modules/libcanberra/libcanberra.json",
         {
             "name" : "evolution-data-server",
             "buildsystem" : "cmake-ninja",
```

**Comment by go-ten** [comment]
Hi,

Any news about this PR ?

**Comment by bilelmoussaoui** [comment]
> > Some module were not updated, e.g. libgweather as the compiler complained.
> 
> What was the issue here? I tried your branch plus libgweather 40.0 locally and it compiles. Granted, the weather part doesn't seem to actually work (GWeather-CRITICAL **: 09:34:32.217: gweather_info_update: assertion 'info->contact_info != NULL' failed in the terminal, and no weather info appears), but otherwise it looks fine to me.

lt depends on whether Calendar was updated to use the new libgweather 40 API or not.

**Comment by X-m7** [comment]
> lt depends on whether Calendar was updated to use the new libgweather 40 API or not.

According to [this](https://gitlab.gnome.org/GNOME/gnome-calendar/-/issues/682) it's been updated enough to compile at least, which it does on my machine.

**Comment by bilelmoussaoui** [comment]
@A6GibKm do you mind bumping libgweather & goa as well?

**Comment by A6GibKm** [comment]
@bilelmoussaoui I am not able to build goa. Seems whatever they do in the flatpak manifest upstream uses master features which are not even in 3.9.

**Comment by X-m7** [comment]
> I am not able to build goa. Seems whatever they do in the flatpak manifest upstream uses master features which are not even in 3.9.

Which version did you try? The latest release is 3.38.1 still, that again compiles fine on my system. There is a 3.40 branch on GItLab but they haven't actually tagged a release from it yet.

**Comment by A6GibKm** [comment]
I tried using 3.8.5 without success.

**Comment by X-m7** [comment]
> I tried using 3.8.5 without success.

Makes sense, that was all the way back from 2013, easy mistake to make if you were just scrolling through https://download.gnome.org/sources/gnome-online-accounts/.

**Comment by A6GibKm** [comment]
I found the issue.... I was using 3.8 rather than 3.38..

File: org.gnome.Calendar.json [hunk 2]
```diff
@@ -46,8 +46,8 @@
             "sources" : [
                 {
                     "type" : "archive",
-                    "url" : "https://download.gnome.org/sources/gnome-online-accounts/3.36/gnome-online-accounts-3.36.0.tar.xz",
-                    "sha256" : "1c8f62990833ca41188dbb80c5e99d99b57a62608ca675bbcd37bc2244742f2e"
+                    "url": "https://download-fallback.gnome.org/sources/gnome-online-accounts/3.38/gnome-online-accounts-3.38.1.tar.xz",
```
Comment (bilelmoussaoui) [inline org.gnome.Calendar.json:49]: Should be download.gnome.org, the fallback is just a mirror

Comment (A6GibKm) [reply org.gnome.Calendar.json]: Lol, sorry that was the link I got by right clicking 'save download link'

**Comment by bilelmoussaoui** [comment]
Lgtm, thanks!

**Pull request merged**